pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -78,6 +79,16 @@ func NewReconciler(mgr manager.Manager, rateLimiter flowcontrol.RateLimiter) *Re
 	}
 }
 
+func indexCDByClusterPool(o client.Object) []string {
+	cd := o.(*hivev1.ClusterDeployment)
+	if cd.Spec.ClusterPoolRef != nil {
+		return []string{
+			cd.Spec.ClusterPoolRef.Namespace + "/" + cd.Spec.ClusterPoolRef.PoolName,
```

Comment:
both the indexer and filter needs to perform this task. so
- we use something that is standard and know like types.NamespacedName{}.String()
- or have a function that creates these keys

Reply:
The indexer is building the key from the CD's ClusterPoolRef, whereas the filter is building it from the pool's namespacedname, so the gain isn't huge, but done.